docs/config.md

```diff
@@ -1,211 +1,466 @@
-   and registrations. If not set then a calculated version will be used.
-1. `custom_translations_url`: An optional URL to allow overriding of translatable strings.
-   The JSON file must be in a format of `{"affected string": {"languageCode": "new string"}}`.
-   See https://github.com/matrix-org/matrix-react-sdk/pull/7886 for details.
-
-Note that `index.html` also has an og:image meta tag that is set to an image
-hosted on riot.im. This is the image used if links to your copy of Element
-appear in some websites like Facebook, and indeed Element itself. This has to be
-static in the HTML and an absolute URL (and HTTP rather than HTTPS), so it's
-not possible for this to be an option in config.json. If you'd like to change
-it, you can build Element, but run
-`RIOT_OG_IMAGE_URL="http://example.com/logo.png" yarn build`.
-Alternatively, you can edit the `og:image` meta tag in `index.html` directly
-each time you download a new version of Element.
-
-Identity servers
-================
+# Configuration
+
+You can configure the app by copying `config.sample.json` to `config.json` and customising it. The file is in JSON
+format and can often result in syntax errors when editing it - visit [#element-web:matrix.org](https://matrix.to/#/#element-web:matrix.org)
+on Matrix if you run into issues. The possible options are described here.
+
+For a good example of a production-tuned config, see https://app.element.io/config.json
+
+For an example of a development/beta-tuned config, see https://develop.element.io/config.json
+
+After changing the config, the app will need to be reloaded. For web browsers this is a simple page refresh, however
+for the desktop app the application will need to be exited fully (including via the task tray) and re-started.
+
+## Homeserver configuration
+
+In order for Element to even start you will need to tell it what homeserver to connect to *by default*. Users will be
+able to use a different homeserver if they like, though this can be disabled with `"disable_custom_urls": false` in your
+config.
+
+One of the following options **must** be supplied:
+
+1. `default_server_config`: The preferred method of setting the homeserver connection information. Simply copy/paste
+   your [`/.well-known/matrix/client`](https://spec.matrix.org/latest/client-server-api/#getwell-knownmatrixclient)
+   into this field. For example:
+   ```json
+   {
+      "default_server_config": {
+         "m.homeserver": {
+            "base_url": "https://matrix-client.matrix.org"
+         },
+         "m.identity_server": {
+            "base_url": "https://vector.im"
+         }
+      }
+   }
+   ```
+2. `default_server_name`: A deprecated method of connecting to the homeserver by looking up the connection information
+   using `.well-known`. When using this option, simply use your server's domain name (the part at the end of user IDs):
+   `"default_server_name": "matrix.org"`
```

Comment:
If this is deprecated we should make this the last option, and maybe add a strike through the property name to discourage people from using it

Reply:
"deprecated", but we still use it :( 

The deprecation is probably false on this tbh.